opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -646,12 +625,7 @@ void EclipseWriter::writeTimeStep(int report_step,
        the conversion is done once here - and not closer to the actual
        use-site.
     */
-    convertFromSiTo( cells[ dc::PRESSURE ], units, UnitSystem::measure::pressure );
-    if ( cells.has( dc::TEMP ))
-        convertFromSiTo( cells[ dc::TEMP ],
-                         units,
-                         UnitSystem::measure::temperature );
-
+    cells.convertFromSI( units );
```

Comment:
Have added a internal bool flag to the container which keeps track of whether the data is in SI units or not.